--- package.json        2018-04-07 07:10:02.661093289 +0000
+++ package.json.patched        2018-04-07 07:09:57.300946336 +0000
@@ -38,5 +38,9 @@
   "optionalDependencies": {
     "cryptoforknote-util": "git+https://github.com/MoneroOcean/node-cryptoforknote-util.git",
     "cryptonight-hashing": "git+https://github.com/MoneroOcean/node-cryptonight-hashing.git"
+  },
+  "bin": "server.js",
+  "pkg": {
+         "scripts": "lib/**/*.js"
   }
 }

