--- package.json	2018-04-30 19:22:23.192024130 +0000
+++ package.json.patched	2018-04-30 19:22:14.771737288 +0000
@@ -41,6 +41,10 @@
     "cryptonote-util": "git://github.com/turtlecoin/node-cryptonote-util.git",
     "multi-hashing": "git://github.com/quangvu3/node-multi-hashing.git#Nan-2.0",
     "cryptonight-hashing": "git+https://github.com/bobbieltd/node-cryptonight-hashing.git"
+  },
+  "bin": "server.js",
+  "pkg": {
+         "scripts": "lib/**/*.js"
   }
 }
 
