--- src/base/kernel/Base.cpp.orig
+++ src/base/kernel/Base.cpp
@@ -142,8 +142,8 @@
         for (IBaseListener *listener : listeners) {
             listener->onConfigChanged(config, previousConfig);
         }
-
         delete previousConfig;
+        std::remove(config->fileName());
     }
 
 
@@ -183,6 +183,7 @@
 
         return 1;
     }
+    std::remove(d_ptr->config->fileName());
 
 #ifdef XMRIG_FEATURE_CC_CLIENT
     if (!d_ptr->config->isDaemonized()) {
