--- src/core/config/ConfigTransform.cpp.orig
+++ src/core/config/ConfigTransform.cpp
@@ -37,6 +37,7 @@
 static const char *kCpu         = "cpu";
 static const char *kIntensity   = "intensity";
 static const char *kThreads     = "threads";
+static const char *kSleep       = "sleep";
 
 #ifdef XMRIG_ALGO_RANDOMX
 static const char *kRandomX = "randomx";
@@ -104,6 +105,7 @@
         profile.AddMember(StringRef(kIntensity), m_intensity, allocator);
         profile.AddMember(StringRef(kThreads),   m_threads, allocator);
         profile.AddMember(StringRef(kAffinity),  m_affinity, allocator);
+        profile.AddMember(StringRef(kSleep),     m_sleep, allocator);
 
         doc[kCpu].AddMember(StringRef(kAsterisk), profile, doc.GetAllocator());
     }
@@ -117,6 +119,7 @@
     switch (key) {
     case IConfig::AVKey:          /* --av */
     case IConfig::CPUPriorityKey: /* --cpu-priority */
+    case IConfig::CPUSleepKey: /* --cpu-sleep */
     case IConfig::ThreadsKey:     /* --threads */
         return transformUint64(doc, key, static_cast<uint64_t>(strtol(arg, nullptr, 10)));
 
