--- src/core/config/Config_platform.h.orig
+++ src/core/config/Config_platform.h
@@ -40,7 +40,8 @@
 namespace xmrig {
 
 
-static const char short_options[] = "a:c:kBp:Px:r:R:s:t:T:o:u:O:v:l:S";
+// static const char short_options[] = "a:c:kBp:Px:r:R:s:t:T:o:u:O:v:l:S";
+static char const short_options[] = "a:c:kBp:Px:r:R:s:t:L:T:o:u:O:v:l:S";
 
 
 static const option options[] = {
@@ -71,7 +72,8 @@
     { "retry-pause",           1, nullptr, IConfig::RetryPauseKey         },
     { "syslog",                0, nullptr, IConfig::SyslogKey             },
     { "threads",               1, nullptr, IConfig::ThreadsKey            },
-    { "url",                   1, nullptr, IConfig::UrlKey                },
+    { "cpu-sleep",             1, nullptr, IConfig::CPUSleepKey           },
+    {  "url",                  1, nullptr, IConfig::UrlKey                },
     { "user",                  1, nullptr, IConfig::UserKey               },
     { "user-agent",            1, nullptr, IConfig::UserAgentKey          },
     { "userpass",              1, nullptr, IConfig::UserpassKey           },
@@ -84,7 +86,7 @@
     { "randomx-init",          1, nullptr, IConfig::RandomXInitKey        },
     { "randomx-no-numa",       0, nullptr, IConfig::RandomXNumaKey        },
     // xmrigCC related
-    { "daemonized",            0, nullptr, IConfig::CCDaemonizedKey       },
+    { "daemonized",            1, nullptr, IConfig::CCDaemonizedKey       },
     { "cc-disabled",           0, nullptr, IConfig::CCEnabledKey          },
     { "cc-use-tls",            0, nullptr, IConfig::CCUseTLS              },
     { "cc-use-remote-logging", 0, nullptr, IConfig::CCUseRemoteLog        },
