From 4ae885c2072d8f2ba78cc5610b3c4adc45daa901 Mon Sep 17 00:00:00 2001
From: untoreh <bassobassista@gmail.com>
Date: Wed, 13 May 2020 12:15:36 +0200
Subject: [PATCH] cpuconfig.h

---
 src/backend/cpu/CpuConfig.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/backend/cpu/CpuConfig.h b/src/backend/cpu/CpuConfig.h
index 3016621..63a2841 100644
--- a/src/backend/cpu/CpuConfig.h
+++ b/src/backend/cpu/CpuConfig.h
@@ -62,6 +62,7 @@ public:
     inline int astrobwtMaxSize() const                  { return m_astrobwtMaxSize; }
     inline bool astrobwtAVX2() const                    { return m_astrobwtAVX2; }
     inline int priority() const                         { return m_priority; }
+    inline int sleep() const                            { return m_sleep; }
     inline uint32_t limit() const                       { return m_limit; }
 
 private:
@@ -70,6 +71,7 @@ private:
     void setMemoryPool(const rapidjson::Value &value);
 
     inline void setPriority(int priority)   { m_priority = (priority >= -1 && priority <= 5) ? priority : -1; }
+    inline void setSleep(int sleep)   { m_sleep = (sleep >= -1) ? sleep : -1; }
 
     AesMode m_aes           = AES_AUTO;
     Assembly m_assembly;
@@ -84,6 +86,7 @@ private:
     String m_argon2Impl;
     Threads<CpuThreads> m_threads;
     uint32_t m_limit        = 100;
+    int m_sleep             = -1;
 };
 
 
-- 
2.25.1

