--- Client.cpp.orig
+++ Client.cpp
@@ -233,29 +233,29 @@
     PowVariant powVariant = Options::i()->powVariant();
 
     if (!Options::i()->forcePowVariant() || m_donate) {
-        if (params.HasMember("algo")) {
-            std::string algo = params["algo"].GetString();
-
-            if (algo.find("/") != std::string::npos) {
-                powVariant = parseVariant(algo.substr(algo.find("/") + 1));
-            }
+      if (params.HasMember("algo")) {
+        std::string algo = params["algo"].GetString();
+        LOG_INFO("algo: %s", algo.c_str());
+        if (algo.find("/") != std::string::npos) {
+          powVariant = parseVariant(algo.substr(algo.find("/") + 1));
         }
+      }
 
-        if (params.HasMember("variant")) {
-            const rapidjson::Value& variant = params["variant"];
+      if (params.HasMember("variant")) {
+        const rapidjson::Value& variant = params["variant"];
 
-            PowVariant parsedVariant = powVariant;
+        PowVariant parsedVariant = powVariant;
 
-            if (variant.IsInt()) {
-                parsedVariant = parseVariant(variant.GetInt());
-            } else if (variant.IsString()) {
-                parsedVariant = parseVariant(variant.GetString());
-            }
+        if (variant.IsInt()) {
+          parsedVariant = parseVariant(variant.GetInt());
+        } else if (variant.IsString()) {
+          parsedVariant = parseVariant(variant.GetString());
+        }
 
-            if (parsedVariant != POW_AUTODETECT) {
-                powVariant = parsedVariant;
-            }
+        if (parsedVariant != POW_AUTODETECT) {
+          powVariant = parsedVariant;
         }
+      }
     }
 
     job.setPowVariant(powVariant);
