--- src/backend/cpu/CpuConfig.h.orig
+++ src/backend/cpu/CpuConfig.h
@@ -58,6 +58,7 @@
     inline const String &argon2Impl() const             { return m_argon2Impl; }
     inline const Threads<CpuThreads> &threads() const   { return m_threads; }
     inline int priority() const                         { return m_priority; }
+    inline int sleep() const                            { return m_sleep; }
 
 private:
     void generate(uint32_t limit);
@@ -65,6 +66,7 @@
     void setAesMode(const rapidjson::Value &aesMode);
 
     inline void setPriority(int priority)   { m_priority = (priority >= -1 && priority <= 5) ? priority : -1; }
+    inline void setSleep(int sleep)   { m_sleep = (sleep >= -1) ? sleep : -1; }
 
     AesMode m_aes        = AES_AUTO;
     Assembly m_assembly;
@@ -72,6 +74,7 @@
     bool m_hugePages     = true;
     bool m_shouldSave    = false;
     int m_priority       = -1;
+    int m_sleep          = -1;
     int32_t  m_limit     = 100;
     String m_argon2Impl;
     Threads<CpuThreads> m_threads;
